gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala

```diff
@@ -0,0 +1,30 @@
+package geotrellis.contrib.vlm.gdal
+
+import geotrellis.contrib.vlm.DataPath
+
+import java.net.MalformedURLException
+
+
+case class GDALDataPath(val path: String) extends DataPath {
+  protected def servicePrefix: String = "gdal+"
+
+  private lazy val strippedPath: String =
```

Comment:
`lazy` is to be avoided without specific reason. here it might delay error checking on the URI until its used somehow which is unhelpful behavior.